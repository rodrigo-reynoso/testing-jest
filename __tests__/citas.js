import Citas from '../js/classes/Citas.js'

describe('Probando la clase de Citas',()=>{
    const cita = new Citas();
    const id = Date.now();
    test('Agregar una nueva cita',()=>{
        const citaObj = {
            id,
            mascota: 'Toribio',
            propietario: 'Lili',
            telefono: '546546546546',
            fecha: '22-04-2022',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }
       
        cita.agregarCita(citaObj);
        // Prueba
        expect(cita).toMatchSnapshot();
    });
    test('Actualizar cita',()=>{
        const citaActualizada = {
            id,
            mascota: 'Toribio actualizado',
            propietario: 'Lili',
            telefono: '546546546546',
            fecha: '22-04-2022',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }
        cita.editarCita(citaActualizada);
        expect(cita).toMatchSnapshot();
    });
    test('Eliminar cita',()=>{
        cita.eliminarCita(id);
        expect(cita).toMatchSnapshot();
    });

});