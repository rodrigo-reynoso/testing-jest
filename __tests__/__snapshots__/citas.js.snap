// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "22-04-2022",
      "hora": "10:30",
      "id": 1650671759769,
      "mascota": "Toribio actualizado",
      "propietario": "Lili",
      "sintomas": "Solo duerme",
      "telefono": "546546546546",
    },
  ],
}
`;

exports[`Probando la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "22-04-2022",
      "hora": "10:30",
      "id": 1650671759769,
      "mascota": "Toribio",
      "propietario": "Lili",
      "sintomas": "Solo duerme",
      "telefono": "546546546546",
    },
  ],
}
`;

exports[`Probando la clase de Citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
